### PR TITLE
feat: add block hash, number and timestamp field to `BlobsOnTransactions` model

### PR DESCRIPTION
**Pull request by PJColombo**
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description

It builds on top of PR #422.

It creates two new fields `block_hash`, `block_number` and `block_timestamp` in the `blobs_on_transactions` table and populates them. It also creates an index for each of them.

#### Motivation and Context (Optional)

This PR addresses the performance issues with the current Prisma query used to retrieve multiple blobs. The existing approach is inefficient because it retrieves full blob data before filtering or sorting, leading to unnecessary overhead.

### CODE REVIEW

File: packages/db/prisma/schema.prisma [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 generator client {
   provider        = "prisma-client-js"
-  previewFeatures = ["tracing", "metrics"]
+  previewFeatures = ["relationJoins", "tracing", "metrics"]
```
Comment (PabloCastellano) [inline packages/db/prisma/schema.prisma:6]: Are we using `relationJoins` in the end?

File: packages/api/test/__snapshots__/blob.test.ts.snap [hunk 2]
```diff
@@ -55,9 +55,6 @@ exports[`Blob router > getAll > when getting expanded blob results > should retu
     "index": 2,
     "proof": "proof004",
     "size": 1300,
-    "transaction": {
-      "rollup": "optimism",
-    },
```
Comment (PabloCastellano) [inline packages/api/test/__snapshots__/blob.test.ts.snap:60]: why is this data being removed?

File: packages/api/test/__snapshots__/blob.test.ts.snap [hunk 3]
```diff
@@ -36,11 +36,11 @@ exports[`Blob router > getAll > when getting expanded blob results > should retu
       "blobGasPrice": "22",
       "blobGasUsed": "5500000",
       "excessBlobGas": "15000",
-      "hash": "blockHash007",
       "slot": 107,
-      "timestamp": "2023-08-31T14:00:00.000Z",
     },
+    "blockHash": "blockHash007",
     "blockNumber": 1007,
+    "blockTimestamp": "2023-08-31T14:00:00.000Z",
```
Comment (PabloCastellano) [inline packages/api/test/__snapshots__/blob.test.ts.snap:43]: Is this a breaking change for the API users? In this case, we should warn it

File: packages/db/prisma/schema.prisma (hunk 1 above)
Comment (PJColombo) [reply packages/db/prisma/schema.prisma]: No, I accidentally left it in while testing. Good catch.

It looks is less performant compared to the current Prisma relation strategy when you try to pull elements from the db with a large offset.

File: packages/api/test/__snapshots__/blob.test.ts.snap (hunk 3 above)
Comment (PJColombo) [reply packages/api/test/__snapshots__/blob.test.ts.snap]: It shouldn't be a breaking change.

We're keeping the current structure and just adding a few extra fields

**Pull request merged**